google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -119,6 +146,175 @@ static void enableApis(Set<CloudLibrary> libraries, CloudProject cloudProject, P
     }
   }
 
+  /**
+   * Creates a new {@link ServiceAccount}, adds the supplied set of {@link Role roles} to it, and
+   * creates and downloads the service account private key to the user's file system.
+   *
+   * @param roles the set of {@link Role} to add to the new service account
+   * @param name the name of the new service account to be created
+   * @param downloadDir the {@link Path} of the download directory of the service account private
+   *     key json file
+   * @param cloudProject the current {@link CloudProject}
+   * @param project the current {@link Project}
+   */
+  static void createServiceAccountAndDownloadKey(
+      Set<Role> roles, String name, Path downloadDir, CloudProject cloudProject, Project project) {
+    Optional<CredentialedUser> user =
+        Services.getLoginService().getLoggedInUser(cloudProject.googleUsername());
+
+    if (!user.isPresent()) {
+      LOG.error("Cannot enable APIs: logged in user not found.");
+      return;
+    }
+
+    ProgressIndicator progress = ProgressManager.getInstance().getProgressIndicator();
+
+    try {
+      int numSteps = roles.isEmpty() ? 3 : 4;
+      double step = 1;
+
+      showProgress(
```

Comment:
It looks like this method is running on thread provided by `ProgressManager.runProcessWithProgressSynchronously()`, could you recall if progress bar allow calling its method from non-UI threads? I.e. `showProgress()`calls `setText()` etc. from here.

Reply:
thats a good question. The platform isn't raising any EDT related exceptions like it typically does so I am assuming that ProgressManager internally handles doing things on the correct thread.

Reply:
Awesone, this makes it much easier to use!